### ISSUE TITLE
Industries Farm buildings not spawning no lane restrictions on road, multiple locations.

### ISSUE THREAD

**Issue by citymangler** [CitiesSkylinesMods/TMPE#494]
Hi,

In this save most of the farm extractors and intermediate (animal enclosures, milking sheds etc) are failing to spawn trucks. They try but the truck despawns almost instantly. The Broken Node Mod cannot detect any issues and Watch It shows that traffic is only 50% of the limit. The only 100% being the number of squares (9).

Others are only spawning a limit number (say 3) of trucks.

Have tried to reset stuck vehicles but this has no effect. Ban private cars from bus lanes is off, and has never been used,  There are no lane restrictions etc in the roads that the assets attach to and I cannot see any obvious reason why a truck could not get to anywhere in the city. Building a bus route around works.

Have tried setting vehicle restriction aggression to low, but again no joy.

The city has some workshop content as assets - 
2 platforms cross diagonal by Cyterion
8 * 6 car park A by Crux Magic
Aqua fun park by bennymedia
Double track cargo station by Bloody Penguin
Roads by rahtazz - Four lanes tram express; four lanes tram express, grass;four lanes tram express,trees;
ifriqiya's - large road with bike and tram, large road with tram tracks,large road with trees and tram
monorail and tram hub by arKur
2 of my own assets (not available on the workshop but included in the zip)

I have never had any problems with the above before now.

Only other Mods I use: Roundabout builder, move it (but not that much), fine road anarchy (again not much) and Pedestrian crossing by Spectra.

I have others subscribed, like Prefab hook and more network stuff by Bloody Penguin -so that I can add stops to stations etc. No idea if these could be causing the issue they have not in the past.

### Steps to reproduce
This started to happen after I rebuilt one farm area. I thought that perhaps a roundabout that the roundabout mod had added might be the cause but removing this does not fix the issue, and as mentioned the broken node mod cannot find a problem.

If I remove TPME then the extractors start spawning but everything else jams, all service vehicles stop, all metros stop. all buses etc stop. It looks like the game has almost forgotten about them, except any other road vehicle sits behind the jammed one and either depawns or eventually crawls past. The stuck vehicles never despawn but deleting their home building will remove them. Unfortunately you cannot do this with trucks etc owned by other cities so eventually everything grinds to a halt and the city dies.

The farm extractor spawn issue occurs in both stable and labs versions of TPME.

Regarding the issues when you try and remove TPME, would it be possible to add an option to disable tpme so it hands back control gracefully?

Hope you can help.

### Log files
[Save and other files.zip ](https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/files/3512090/Save.and.other.files.zip)

**Comment by aubergine10** [comment]
Thanks for the detailed error report!

Could you upload the log files too please - [`output_log.txt` and `TMPE.log`](https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/wiki/Share-your-Cities-Skylines-log-file).

> Regarding the issues when you try and remove TPME, would it be possible to add an option to disable tpme so it hands back control gracefully?

TM:PE doesn't alter the save game so it should already remove gracefully. Only thing I can think of is the paths stored in the vehicles that were on map at time of saving, but those paths are in vanilla format so shouldn't cause a problem.

> @krzychu124 Is my statement above about paths correct? If memory serves the parking AI kicks in separately, it's not part of the path?

**Comment by citymangler** [comment]
Hi,
Many thanks for such a quick reply. Here's a zip with the logs as well as a save with the broken city - after TPME is removed.

Looking through the game log shows that something is stating that the 'Broken Node Detector disabled' Could this be from the broken node mod, or could this be the issue? The line is at the very end of the output log.

[logs and broken city save.zip](https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/files/3515241/logs.and.broken.city.save.zip)

**Comment by aubergine10** [comment]
Log files don't indicate any notable issues, all TM:PE patches/detours seem to be applied correctly.

As for the `Broken Nodes Detector disabled` message, that's normal operation - it outputs that when the game exits to confirm its correctly disengaged.

Save loads, just need to sub some missing road assets and then I should be able to do more extensive testing.

**Comment by aubergine10** [comment]
Just making note of assets that I'm subbing to test the save...

### Networks

* 3 Lane oneway road -- can't find in workshop (maybe it's NExt2?)
* [Four lanes, tram express](https://steamcommunity.com/sharedfiles/filedetails/?id=1512210223)
* [Four lanes, tram express, grass](https://steamcommunity.com/sharedfiles/filedetails/?id=1512213893)
* [Four lanes, tram express, trees](https://steamcommunity.com/sharedfiles/filedetails/?id=1512214569)
* [HPO1](https://steamcommunity.com/sharedfiles/filedetails/?id=1580457997)
* [Large Road Elevated0](https://steamcommunity.com/sharedfiles/filedetails/?id=1512214569)
* [Large Road with Bike and Tram](https://steamcommunity.com/sharedfiles/filedetails/?id=1226048259)
* [Large Road with Tram Tracks](https://steamcommunity.com/sharedfiles/filedetails/?id=1195353447)
* [Large Road with Trees and Tram](https://steamcommunity.com/sharedfiles/filedetails/?id=1196140880)
* [Large Road with Trees, Bus, Tram](https://steamcommunity.com/sharedfiles/filedetails/?id=1196195846)
* Medium Road Elevated Tram0 - can't find in workshop (NExt2?)
* [Pedestrian Elevated0](https://steamcommunity.com/sharedfiles/filedetails/?id=1580457997)
* Tram and Monorail road - can't find in workshop (NExt2?)
* Tram stop on 4 ln road - can't find in workshop (NExt2?)

### Buildings

* [2 Platforms Cross Diagonal](https://steamcommunity.com/sharedfiles/filedetails/?id=1184623662)
* ~~8x6 Carpark A~~ - skipped
* Cogvos' Dual line Metro - could not find in workshop
* [Double Track Cargo Station](https://steamcommunity.com/sharedfiles/filedetails/?id=703597323)
* ~~HighwayBridgePillar~~ - skipped
* [HPOpier01](https://steamcommunity.com/sharedfiles/filedetails/?id=1580457997) - ped bridge
* [Integrated Metro Station](https://steamcommunity.com/sharedfiles/filedetails/?id=942726934)
* Integrated Metro Station - duplicate?
* [Monorail and Tram Hub](https://steamcommunity.com/sharedfiles/filedetails/?id=942726934)

**Comment by aubergine10** [comment]
Location of `Screenshot 35.png` from zip file in OP:

![image](https://user-images.githubusercontent.com/1386719/63274016-f43c8280-c296-11e9-9621-bd80bc85642d.png)

The missing workshop roads mentioned in my previous comment are preventing accurate testing, for example;

![image](https://user-images.githubusercontent.com/1386719/63274660-1aaeed80-c298-11e9-8ba6-25c71238e764.png)

![image](https://user-images.githubusercontent.com/1386719/63274759-4336e780-c298-11e9-9263-770e843c47e8.png)

![image](https://user-images.githubusercontent.com/1386719/63275197-fbfd2680-c298-11e9-8386-481738215e65.png)

If you can identify any of the missing assets (those mentioned in my earlier comment) and post links to them it would help with testing.

**Comment by aubergine10** [comment]
There are some junctions with partial lane connectors (some lanes don't have lane connectors, others do), for example:

![image](https://user-images.githubusercontent.com/1386719/63275538-9a898780-c299-11e9-8eab-fb313618e91c.png)

Not sure if that causes problems but I vaguely remember LinuxFan mentioning that if lane connectors are used all lanes must be defined.

**Comment by aubergine10** [comment]
Noticed this in my log file after testing the save:

```csharp
Simulation error: Object reference not set to an instance of an object
at (wrapper dynamic-method) VehicleManager.CreateVehicle_Patch2 (object,uint16&,ColossalFramework.Math.Randomizer&,VehicleInfo,UnityEngine.Vector3,TransferManager/TransferReason,bool,bool) <0x00ae3>
at RocketLaunchAI.SimulationStep (uint16,EventData&) <0x0083d>
at EventManager.SimulationStepImpl (int) <0x00149>
at SimulationManagerBase`2.SimulationStep (int) <0x0004c>
at EventManager.ISimulationManager.SimulationStep (int) <0x00024>
at SimulationManager.SimulationStep () <0x00693>
at SimulationManager.SimulationThread () <0x0018a>
  [Core]
```

And on game exit this:

```csharp
NullReferenceException: Object reference not set to an instance of an object
  at TrafficManager.UI.TrafficManagerTool.OnEnable () [0x00000] in <filename unknown>:0 
UnityEngine.GameObject:Internal_AddComponentWithType(Type)
UnityEngine.GameObject:AddComponent(Type)
UnityEngine.GameObject:AddComponent()
TrafficManager.UI.UIBase:GetTrafficManagerTool(Boolean)
TrafficManager.UI.UIBase:EnableTool()
TrafficManager.UI.UIBase:SetToolMode(TrafficManagerMode)
TrafficManager.UI.UIBase:Show()
TrafficManager.UI.UIBase:ToggleMainMenu()
TrafficManager.UI.UIMainMenuButton:OnClick(UIMouseEventParameter)
ColossalFramework.UI.UIComponent:OnClick(UIMouseEventParameter)
ColossalFramework.UI.MouseHandler:ProcessInput(IInputTranslator, Ray, UIComponent, Boolean)
ColossalFramework.UI.UIInput:ProcessMouseInput()
ColossalFramework.UI.UIInput:Update()
```

Note: I was using 11.0-alpha6 at the time

**Comment by citymangler** [comment]
Hi,

Many thanks for the reply. To help with the missing assets;

Cogvos' Dual line Metro - could not find in workshop - This should have been in the be in the 1st Zip file I submitted - its an asset I put together but did not release to the workshop as, although it works I did not feel it was good enough. Unfortunately it looks like I attached the wrong version in the 1st zip, please see the new one. Sorry about that.

[Cogvos dual line metro.zip](https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/files/3522402/Cogvos.dual.line.metro.zip

Tram and Monorail road - can't find in workshop (NExt2?) - another one of my efforts using the editor, should be in the 1st zip.

Medium Road Elevated Tram0 - can't find in workshop (NExt2?) - there is only (as far as I can tell) one elevated road with a tram, near a multiplatform train station. Again this actually a tram and monorail road - just over a bridge and the file is in the 1st zip.

3 Lane oneway road -- can't find in workshop (maybe it's NExt2?) - sorry my bad here I completely forgot that I made this(?) or maybe I didn't and got it somewhere else(???) anyway here it is in a zip

[Three lane oneway road.zip](https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/files/3522466/Three.lane.oneway.road.zip)

Tram stop on 4 ln road - can't find in workshop (NExt2?) - not sure on this, again it could be caused by the tram and monorail road I built, if so please check the 1st zip its called - Tram and monorail road.crp If this does not fix then please let me know and I will investigate further.

I don't unfortunately know about the duplicate integrated metro, unless its a goof from the dual metro station I made.

Regarding the errors, I am not using any workshop vehicles so I am not sure what that could be(?) the rocket is under construction but has not yet spawned.

Had no idea on the possible issue re not having lane connectors on all of a node lanes.

Anyway hope this helps.

**Comment by aubergine10** [comment]
@citymangler Do you have asset skipping file set in mod options for Loading Screen Mod? If so could you attach it? Just drag the txt file on to a comment box here and wait for `[Uploading ...]` text to change then click Comment button; no need to put it in a zip.

**Comment by citymangler** [comment]
Hi,

I haven't been using the Loading Screen Mod - didn't know i existed. Anyway installed it and it produced this in the text file. its so small that I just copied the text

#As you know, LSM reports missing and duplicate assets on the loading screen and in the Assets report.
#Using this file, you can stop reporting some assets if you like.

#LSM reported these assets. To stop reporting them, just remove the hashtags.

#Reported missing.
#484381291.Bush #4 (by Breeze)_Data
#512781837.Bush #7 (by Breeze)_Data

According to the HTML (attached in zip) 
[destroy cargo hubs - Assets Report-2019-08-26_10-55-36.zip](https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/files/3540625/destroy.cargo.hubs.-.Assets.Report-2019-08-26_10-55-36.zip)

These are used in the Car park asset - seems unlikely that that is the cause(?)

**Comment by wabuilderman** [comment]
Hey, just as another user who had a similar issue, the partial lane-connectors definitely seemed to have been my issue.

**Comment by maxtraxx** [comment]
Created an account to leave a comment here. I had the same issue. Really bizarre, only farm extractor buildings not spewing trucks.

*** Solved for me -- I had too much zone farming industry (as in yellow industry then zoned as farming). I de-zoned about half and hey presto. A massive farm truck traffic jam. :))

Hope that helps

**Comment by aubergine10** [comment]
An issue was recently found where Industries pathfinding wasn't working properly outside the central 25 tile area when using the 81 Tiles mod. A beta test containing a fix for that issue is available in the workshop: [TMPE 11.6 TEST](https://steamcommunity.com/sharedfiles/filedetails/?id=2489276785).
